eth/db/header.py

```diff
@@ -46,6 +50,91 @@ class HeaderDB(HeaderDatabaseAPI):
     def __init__(self, db: AtomicDatabaseAPI) -> None:
         self.db = db
 
+    def get_chain_integrity_info(self) -> Tuple[Tuple[int, int], ...]:
+        return self._get_chain_integrity_info(self.db)
+
+    @classmethod
+    def _get_chain_integrity_info(cls, db: DatabaseAPI) -> Tuple[Tuple[int, int], ...]:
+        try:
+            encoded_integrity_info = db[SchemaV1.make_header_chain_integrity_info_lookup_key()]
+        except KeyError:
+            return ()
+        else:
+            return decode_chain_integrity_info(encoded_integrity_info)
+
+    @classmethod
+    def _update_chain_integrity_info(
```

Comment:
The code below could move into a shared place and be used to track the chain integrity of blocks in the same way.